### PR TITLE
[WFLY-13352] Add Galleon layer for SAR subsystem

### PR DESCRIPTION
**Pull request by yersan**
Jira issue: https://issues.redhat.com/browse/WFLY-13352

### CODE REVIEW

File: sar/WFLY-13352-Add_galleon_layer_sar_subsystem.adoc [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+The test coverage of the Galleon layer added by this proposal is divided in two main groups:
+
+. Testing the Galleon layer provisioning. This testing is done by https://github.com/wildfly/wildfly/blob/master/testsuite/layers/src/test/java/org/jboss/as/test/layers/LayersTestCase.java[LayersTestCase]. The testsuite will be modified to add a new server provisioned with this layer in isolation and with this layer combined with all the layers. For each kind of provisioning, this test does the following:
+
+.. Verifies the provisioned modules are the expected ones.
+.. Verifies the provisioned server starts successfully.
+
+. Execution of WildFly tests related to the feature provisioned by this layer. Reuse the existing tests available on the WildFly test suite, which are directly testing this layer functionalities, and execute them on a server installation provisioned with this layer.
+== Community Documentation
```
Comment (fabiobrz) [inline sar/WFLY-13352-Add_galleon_layer_sar_subsystem.adoc:94]: A line is missing here and causes bad rendering.

File: sar/WFLY-13352-Add_galleon_layer_sar_subsystem.adoc [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+WildFly
+
+=== Other Interested Projects
+
+N/A
+
+== Requirements
+
+* The Galleon layer must provide the sar subsystem.
+* Since the sar subsystem uses services configured by the JMX subsystem, we need to depend on JMX Galleon layer which supplies the JMX subsystem.
```
Comment (fabiobrz) [inline sar/WFLY-13352-Add_galleon_layer_sar_subsystem.adoc:70]: This is just an internal dependency right?
I mean the user is free not to have JMX layer in the config, correct?

**Review by fabiobrz — CHANGES_REQUESTED** [review]
Hi @yersan - thanks for your work. I've dropped a couple of comments, let me know what you think.

File: sar/WFLY-13352-Add_galleon_layer_sar_subsystem.adoc (hunk 2 above)
Comment (yersan) [reply sar/WFLY-13352-Add_galleon_layer_sar_subsystem.adoc:70]: Correct.

The JMX layer is an internal dependency on the SAR Galleon layer. If a user is provisioning SAR Galleon layer, the JMX will be brought in automatically as a transitive dependency.

Comment (fabiobrz) [reply sar/WFLY-13352-Add_galleon_layer_sar_subsystem.adoc:70]: ok thanks

File: sar/WFLY-13352-Add_galleon_layer_sar_subsystem.adoc (hunk 1 above)
Comment (yersan) [reply sar/WFLY-13352-Add_galleon_layer_sar_subsystem.adoc:95]: @fabiobrz Fixed

**Review by fabiobrz — APPROVED** [review]

**Pull request merged**